lib/twilio-ruby/rest/client.rb

```diff
@@ -71,6 +72,11 @@ def initialize(username=nil, password=nil, account_sid=nil, region=nil, http_cli
       # Makes a request to the Twilio API using the configured http client
       # Authentication information is automatically added if none is provided
       def request(host, port, method, uri, params={}, data={}, headers={}, auth=nil, timeout=nil)
+        whitelisted_numbers = ENV.fetch("WHITELISTED_NUMBERS", "").split
+        environment = ENV.fetch("RAILS_ENV", "production")
+        if environment != "production" && !whitelisted_numbers.include?(data["To"])
+          raise WhitelistedError, "'To' number is not whitelisted"
```

Comment:
We would then need to catch this error everywhere, which is a pain :D 
I would expect, that non-whitelisted number is just returning the default response, kind of :ok and that's it, but not failing.

Reply:
The problem is that in all instances, the gem tries to build a entity from the response. Example when sending a message. A message entity. So that would mean we would have to cater for every time of entity the gem has

Reply:
But with the raised error this forked twilio client doesn't fit our needs 🤷‍♀️ 
The flows will be failing, while we're expecting them not to fail, but just mock the real call to Twilio.

Reply:
Okay let me think if there is a way that we can resolve this

Reply:
@iryna-klara just tested and this seems to work 🎉 
https://github.com/klara/twilio-ruby/pull/2/files#diff-83907799c2ed4ae123875acc335cd0891f68ecab8572a1d9c71865e85ca916c7R78

This way we don't need to make a error